Keyspace-cloud/android
# 19 digit cards

## :recycle: Current situation

App doesn't support 19 digit credit cards (like Maestro)

## :bulb: Proposed solution

- Add support for adding 19-digit cards.
- Update app UX to support 19-digit cards.

## 📋 Release Notes

- Added support for 19-digit cards.
- Fix dashboard card to display more digits by moving them to the next line instead of showing ellipses. 
- Add Maestro card to payment gateway detector.
- Updated validation for 16 to 19 digit cards.

## :heavy_plus_sign: Additional Information

#44

## 📝 Testing

1. Try saving a 16 digit card. Observe usual spacing behavior.
2. Add 3 more digits to the card number. Observe how the spacing vanishes.
3. Remove the spacing from the end. Observe how 16-digit spacing reappears.
4. Save a card and try editing it. Notice how the card number formatting is preserved visually.
5. When testing 19-digit cards on narrow length devices (E.g.: the front display of a Samsung Galaxy Fold, notice how the card information is multiline).
